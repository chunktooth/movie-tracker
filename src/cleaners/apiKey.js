export const apiKey = '480eafc74b027454d89765098061cb64';

// jared - 480eafc74b027454d89765098061cb64
// Matt - 4eabf2c86624eba27b3bed0627180793
// om - e7f886548071b04cf8febc293d108511