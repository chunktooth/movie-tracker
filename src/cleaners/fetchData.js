import { apiKey } from './apiKeys';

const rootUrl = 'https://api.themoviedb.org/3';
const nowPlaying = `/movie/now_playing?page=1&language=en-US&api_key=${apiKey}&language=en-US`;

export const getNowPlaying = async () => {

  try {
    const response = await fetch(`${rootUrl}${nowPlaying}`);
    const nowPlayingData = await response.json();

    return nowPlayingData;
  } catch (error) {
    throw new Error ('its bad');
  }
};


export const addToFavorites = async (movie) => {

  try {
    const response = await fetch('/api/users/favorites/new', {
      method: 'POST',
      body: JSON.stringify(movie),
      headers: {
        'content-type': 'application/json'
      }
    });
    const favoriteData = await response.json();

    return favoriteData;
  } catch (error) {
    console.log(error);
  }
}

export const removeFromFavorites = async (userId, movieId) => {

  try {
    const response = await fetch(`/api/users/${userId}/favorites/${movieId}`, {
      method: 'DELETE',
      headers: {
        'content-type': 'application/json'
      }
    });
    const removedData = await response.json();
    
    return removedData;
  } catch (error) {
    console.log(error);
  }
}