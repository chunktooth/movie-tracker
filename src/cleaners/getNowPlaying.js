import { apiKey } from './apiKeys';

export const getNowPlaying = async () => {

  try {
    const response = await fetch(`https://api.themoviedb.org/3/movie/now_playing?page=1&language=en-US&api_key=${apiKey}&language=en-US`);
    const nowPlayingData = await response.json();
    const cleanedMovies = cleanMovies(nowPlayingData)
    return cleanedMovies
  } catch (error) {
    throw new Error ('Cannot fetch');
  }
};

const cleanMovies = (movieData) => {
  return movieData.results.map(movie => ({
    movie_id: movie.id, 
    title: movie.title, 
    poster_path: movie.poster_path, 
    release_date: movie.release_date, 
    vote_average: movie.vote_average, 
    overview: movie.overview  
  }));
}