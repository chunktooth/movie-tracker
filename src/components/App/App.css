.log-out {
  position: absolute;
  right: 10px;
}