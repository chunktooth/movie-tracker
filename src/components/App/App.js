import React, { Component } from 'react';
import './App.css';
import { Route, Switch, NavLink, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { Home } from '../Home/Home';
import { addToFavorites } from '../../cleaners/addToFavorites';
import Login from '../../containers/Login/Login';
import { loadMovies, logOutUser, addFavorite, clearFavoritesLogOut } from '../../actions';
import SignUp from '../SignUp/SignUp.js';
import { getNowPlaying } from '../../cleaners/getNowPlaying';
import { loadAllFavorites } from '../../cleaners/loadAllFavorites';

export class App extends Component {
  
  async componentDidMount() {
    const nowPlaying = await getNowPlaying();

    this.props.loadMovies(nowPlaying);
  }

  displayLogIn = () => {
    return (
      <NavLink 
        to='/login'
        className='nav'> 
        Log In / Sign Up
      </NavLink>
    );
  };

  displayLogOut = () => {
    const { pathname } = this.props.location;

    return (
      <div className='nav-menu'>
      <h3>{`Hey, ${this.props.user.name} !`}</h3>
       <NavLink
          to='/' 
          className='log-out'
          onClick={this.handleLogOut}>
          Log Out
        </NavLink>
        { pathname === '/' ? this.showFavorites() : this.showHome() }
      </div>
    );
  };

  showFavorites = () => {
    return ( 
      <NavLink 
        to='/favorites' 
        className='show-favorites'>
        Show Favorites
      </NavLink>  
    )
  }

  showHome = () => {
    return (
      <NavLink to='/'>Home</NavLink>
    )
  }


  handleLogOut = () => {
    this.props.logOutUser();
    this.props.clearFavoritesLogOut();
  }

  render() {
    const { user, movies, favorites } = this.props;

    return (
      <div className="App">
        <header>
        { !user.id ? this.displayLogIn() : this.displayLogOut()}
        <h1>Movie Tracker</h1>
        </header>

        <Switch>
          <Route exact path='/' 
            render={ () => <Home 
              movies={movies} 
              favorites={favorites} />} 
          />
          <Route exact path='/login'
            component={Login} 
          />
          <Route exact path='/signup'
            component={SignUp}
          />
          <Route exact path='/favorites'
            render={ () => <Home 
              movies={favorites} 
              favorites={favorites} />}
          />
        </Switch>
      </div>
    );
  }
}

export const mapStateToProps = (state) => {
  return {
    user: state.user,
    movies: state.movies,
    favorites: state.favorites
  };
};

export const mapDispatchToProps = (dispatch) => {
  return {
    loadMovies: (movies) => (dispatch(loadMovies(movies))),
    logOutUser: () => (dispatch(logOutUser())),
    clearFavoritesLogOut: () => (dispatch(clearFavoritesLogOut()))
  };
};

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(App));
