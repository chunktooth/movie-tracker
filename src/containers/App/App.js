import React, { Component } from 'react';
import './App.css';
import { Route, Switch, NavLink, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { Home } from '../../components/Home/Home';
import { addToFavorites } from '../../cleaners/addToFavorites';
import Login from '../Login/Login';
import { loadMovies, logOutUser, addFavorite, clearFavoritesLogOut } from '../../actions';
import SignUp from '../../components/SignUp/SignUp.js';
import { getNowPlaying } from '../../cleaners/getNowPlaying';
import { loadAllFavorites } from '../../cleaners/loadAllFavorites';

export class App extends Component {
  
  async componentDidMount() {
    const nowPlaying = await getNowPlaying();

    this.props.loadMovies(nowPlaying);
  }

  displayLogIn = () => {
    return (
      <NavLink 
        to='/login'
        id='login-logout'> 
        Log In / Sign Up
      </NavLink>
    );
  };

  displayLogOut = () => {
    const { pathname } = this.props.location;

    return (
      <div className='nav-menu'>
        <h3 className='welcome'>{`Hello ${this.props.user.name}. We are tracking you....`}</h3>
         <NavLink
            to='/' 
            id='login-logout'
            onClick={this.handleLogOut}>
            Log Out
          </NavLink>
        { pathname === '/' ? this.showFavorites() : this.showHome() }
      </div>
    );
  };

  showFavorites = () => {
    return ( 
      <NavLink 
        to='/favorites' 
        id='show-favorites'>
        Show Favorites
      </NavLink>  
    )
  }

  showHome = () => {
    return (
      <NavLink id='home' to='/'>Home</NavLink>
    )
  }

  handleLogOut = () => {
    this.props.logOutUser();
    this.props.clearFavoritesLogOut();
  }

  render() {
    const { user, movies, favorites } = this.props;

    return (
      <div className="App">
        <header>
        { !user.id ? this.displayLogIn() : this.displayLogOut()}
        <h1>Movie Tracker</h1>
        </header>

        <Switch>
          <Route exact path='/' 
            render={ () => <Home 
              movies={movies} 
              favorites={favorites} />} 
          />
          <Route exact path='/login'
            component={Login} 
          />
          <Route exact path='/signup'
            component={SignUp}
          />
          <Route exact path='/favorites'
            render={ () => <Home 
              movies={favorites} 
              favorites={favorites} />}
          />
        </Switch>
      </div>
    );
  }
}

export const mapStateToProps = (state) => {
  return {
    user: state.user,
    movies: state.movies,
    favorites: state.favorites
  };
};

export const mapDispatchToProps = (dispatch) => {
  return {
    loadMovies: (movies) => (dispatch(loadMovies(movies))),
    logOutUser: () => (dispatch(logOutUser())),
    clearFavoritesLogOut: () => (dispatch(clearFavoritesLogOut()))
  };
};

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(App));
