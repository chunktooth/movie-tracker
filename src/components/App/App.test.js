import React from 'react';
import ReactDOM from 'react-dom';
import { App } from './App';
import { mockNowPlaying } from '../../cleaners/mockData';
import { shallow } from 'enzyme';

describe('App', () => {
  let wrapper;
  let mockUser = [];

  beforeEach(() => {
    wrapper = shallow(<App user={mockUser}/>, { disableLifecycleMethods: true });
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should display log out when user is logged in', () => {
    let mockUser = [{name: 'jared'}];

    wrapper = shallow(<App user={mockUser}/>, { disableLifecycleMethods: true });

    expect(wrapper).toMatchSnapshot();
  });

  // describe('componentDidMount', () => {
  //   it.skip('should invoke getNowPlaying at componentDidMount', () => {
  //     const spy = jest.spyOn();

  //   });

  //   it.skip('should call loadMovies with nowPlaying movie results', () => {
  //     const mockloadMovies = jest.fn();
  //     const mockUsers = [{user: ''}];
  //     const mockgetNowPlaying = jest.fn();

  //     wrapper = shallow(<App 
  //       user={mockUsers}
  //       loadMovies={mockloadMovies} />, {
  //         disableLifecycleMethods: true
  //       });

  //     console.log(wrapper.debug());
  //     wrapper.instance().componentDidMount();
  //     expect(mockloadMovies).toHaveBeenCalled();
  //   });

  // });


  it.skip('should ', () => {

  });

});




