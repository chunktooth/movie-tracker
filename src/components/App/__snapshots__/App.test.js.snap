// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should display log out when user is logged in 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    user={
        Object {
            "id": 4,
            "name": "mud",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header>
          <div>
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              className="log-out"
                              onClick={[Function]}
                              to="/"
                    >
                              Log Out
                    </NavLink>
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              className="nav"
                              to="/favorites"
                    >
                              Show Favorites
                    </NavLink>
          </div>
          <h1>
                    Movie Tracker
          </h1>
</header>,
        <Switch>
          <Route
                    exact={true}
                    path="/"
                    render={[Function]}
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/login"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/signup"
          />
          <Route
                    exact={true}
                    path="/favorites"
                    render={[Function]}
          />
</Switch>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            className="log-out"
                            onClick={[Function]}
                            to="/"
              >
                            Log Out
              </NavLink>
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            className="nav"
                            to="/favorites"
              >
                            Show Favorites
              </NavLink>
</div>,
            <h1>
              Movie Tracker
</h1>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  className="log-out"
                  onClick={[Function]}
                  to="/"
>
                  Log Out
</NavLink>,
                <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  className="nav"
                  to="/favorites"
>
                  Show Favorites
</NavLink>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "ariaCurrent": "true",
                  "children": "Log Out",
                  "className": "log-out",
                  "onClick": [Function],
                  "to": "/",
                },
                "ref": null,
                "rendered": "Log Out",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "ariaCurrent": "true",
                  "children": "Show Favorites",
                  "className": "nav",
                  "to": "/favorites",
                },
                "ref": null,
                "rendered": "Show Favorites",
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Movie Tracker",
            },
            "ref": null,
            "rendered": "Movie Tracker",
            "type": "h1",
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              exact={true}
              path="/"
              render={[Function]}
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/login"
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/signup"
/>,
            <Route
              exact={true}
              path="/favorites"
              render={[Function]}
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/login",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/signup",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/favorites",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header>
            <div>
                        <NavLink
                                    activeClassName="active"
                                    ariaCurrent="true"
                                    className="log-out"
                                    onClick={[Function]}
                                    to="/"
                        >
                                    Log Out
                        </NavLink>
                        <NavLink
                                    activeClassName="active"
                                    ariaCurrent="true"
                                    className="nav"
                                    to="/favorites"
                        >
                                    Show Favorites
                        </NavLink>
            </div>
            <h1>
                        Movie Tracker
            </h1>
</header>,
          <Switch>
            <Route
                        exact={true}
                        path="/"
                        render={[Function]}
            />
            <Route
                        component={[Function]}
                        exact={true}
                        path="/login"
            />
            <Route
                        component={[Function]}
                        exact={true}
                        path="/signup"
            />
            <Route
                        exact={true}
                        path="/favorites"
                        render={[Function]}
            />
</Switch>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                <NavLink
                                activeClassName="active"
                                ariaCurrent="true"
                                className="log-out"
                                onClick={[Function]}
                                to="/"
                >
                                Log Out
                </NavLink>
                <NavLink
                                activeClassName="active"
                                ariaCurrent="true"
                                className="nav"
                                to="/favorites"
                >
                                Show Favorites
                </NavLink>
</div>,
              <h1>
                Movie Tracker
</h1>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="log-out"
                    onClick={[Function]}
                    to="/"
>
                    Log Out
</NavLink>,
                  <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="nav"
                    to="/favorites"
>
                    Show Favorites
</NavLink>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active",
                    "ariaCurrent": "true",
                    "children": "Log Out",
                    "className": "log-out",
                    "onClick": [Function],
                    "to": "/",
                  },
                  "ref": null,
                  "rendered": "Log Out",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active",
                    "ariaCurrent": "true",
                    "children": "Show Favorites",
                    "className": "nav",
                    "to": "/favorites",
                  },
                  "ref": null,
                  "rendered": "Show Favorites",
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Movie Tracker",
              },
              "ref": null,
              "rendered": "Movie Tracker",
              "type": "h1",
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                exact={true}
                path="/"
                render={[Function]}
/>,
              <Route
                component={[Function]}
                exact={true}
                path="/login"
/>,
              <Route
                component={[Function]}
                exact={true}
                path="/signup"
/>,
              <Route
                exact={true}
                path="/favorites"
                render={[Function]}
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/login",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/signup",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/favorites",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;

exports[`App should match the snapshot without a user 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    user={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="nav"
                    to="/login"
          >
                    Log In / Sign Up
          </NavLink>
          <h1>
                    Movie Tracker
          </h1>
</header>,
        <Switch>
          <Route
                    exact={true}
                    path="/"
                    render={[Function]}
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/login"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/signup"
          />
          <Route
                    exact={true}
                    path="/favorites"
                    render={[Function]}
          />
</Switch>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="nav"
              to="/login"
>
              Log In / Sign Up
</NavLink>,
            <h1>
              Movie Tracker
</h1>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Log In / Sign Up",
              "className": "nav",
              "to": "/login",
            },
            "ref": null,
            "rendered": "Log In / Sign Up",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Movie Tracker",
            },
            "ref": null,
            "rendered": "Movie Tracker",
            "type": "h1",
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              exact={true}
              path="/"
              render={[Function]}
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/login"
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/signup"
/>,
            <Route
              exact={true}
              path="/favorites"
              render={[Function]}
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/login",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/signup",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/favorites",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav"
                        to="/login"
            >
                        Log In / Sign Up
            </NavLink>
            <h1>
                        Movie Tracker
            </h1>
</header>,
          <Switch>
            <Route
                        exact={true}
                        path="/"
                        render={[Function]}
            />
            <Route
                        component={[Function]}
                        exact={true}
                        path="/login"
            />
            <Route
                        component={[Function]}
                        exact={true}
                        path="/signup"
            />
            <Route
                        exact={true}
                        path="/favorites"
                        render={[Function]}
            />
</Switch>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="nav"
                to="/login"
>
                Log In / Sign Up
</NavLink>,
              <h1>
                Movie Tracker
</h1>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Log In / Sign Up",
                "className": "nav",
                "to": "/login",
              },
              "ref": null,
              "rendered": "Log In / Sign Up",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Movie Tracker",
              },
              "ref": null,
              "rendered": "Movie Tracker",
              "type": "h1",
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                exact={true}
                path="/"
                render={[Function]}
/>,
              <Route
                component={[Function]}
                exact={true}
                path="/login"
/>,
              <Route
                component={[Function]}
                exact={true}
                path="/signup"
/>,
              <Route
                exact={true}
                path="/favorites"
                render={[Function]}
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/login",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/signup",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/favorites",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
